python/libcudf/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+# =============================================================================
+# Copyright (c) 2022-2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.26.4 FATAL_ERROR)
+
+include(../../rapids_config.cmake)
+
+project(
+  libcudf-python
+  VERSION "${RAPIDS_VERSION}"
+  LANGUAGES CXX
+)
+
+# Check if cudf is already available. If so, it is the user's responsibility to ensure that the
+# CMake package is also available at build time of the Python cudf package.
+find_package(cudf "${RAPIDS_VERSION}")
+
+if(cudf_FOUND)
+  return()
+endif()
+
+unset(cudf_FOUND)
+
+# For wheels, this should always be true
+set(USE_LIBARROW_FROM_PYARROW ON)
+
+# Find Python early so that later commands can use it
+find_package(Python 3.9 REQUIRED COMPONENTS Interpreter)
+
+set(BUILD_TESTS OFF)
+set(BUILD_BENCHMARKS OFF)
+set(CUDF_BUILD_TESTUTIL OFF)
+set(CUDF_BUILD_STREAMS_TEST_UTIL OFF)
+set(CUDA_STATIC_RUNTIME ON)
+
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+
+include(../pylibcudf/cmake/Modules/LinkPyarrowHeaders.cmake)
+
+add_subdirectory(../../cpp cudf-cpp)
+
+# Ensure other libraries needed by libcudf.so get installed alongside it.
+#
+# TODO: This install is currently overzealous. We should only install the libraries that are
```

Comment:
Is this TODO for this PR or a later one?

Reply:
A later one.

This is showing up in the diff here only because this PR is moving some code from `cudf` / `pylibcudf` into this new `libcudf` package.

But it's been in `cudf` since December 2023 (#14565).

I think this comment is still relevant. I'd expected `auditwheel repair` run in wheel builds to make this unnecessary, but I think @vyasr ran into something that made that not quite true: https://github.com/rapidsai/cudf/pull/14565#discussion_r141455416

@vyasr could you look at this and let me know if you agree that this comment + code is still relevant in a `libcudf`-as-a-separate-wheel world?

Reply:
This comment is referring to an edge case that is going to be even more of an edge case going forward and I would be fine simply ignoring that case and deleting this comment. Prior to this PR, running `pip install .` in the python/cudf directory would trigger a build of the Python package. If libcudf could not be found, it would then build libcudf as well. In the default case, nvcomp would also not be present on the system, so we'd download the binaries via rapids-cmake and install those alongside everything else. The edge case is regarding what to do if nvcomp was already installed on the system when building cudf. If nvcomp were already present in a system install directory, technically the line here would be incorrect because we would not want to copy the system-installed nvcomp into the install tree for the Python package. However, I assumed that case would be basically nonexistent in practice and so I just left this as a TODO.

In a libcudf wheel world, the case becomes even less likely. The Python packages (cudf and pylibcudf) will now have a hard dependency on the libcudf Python package being installed and will never trigger a build of libcudf, so the question becomes whether nvcomp already being installed is a relevant case for libcudf builds. Practically speaking, I think that we can assume that nobody is going to be pip installing libcudf from source except us, and since we control all the build environments of note, we can safely assume that nvcomp will not already be present and we can do this install unconditionally.

Reply:
Ok yeah, thanks very much for the explanation, this makes sense to me!

I've removed the TODO part of this comment in https://github.com/rapidsai/cudf/pull/15483/commits/4ba53d84bcfa337baae4baff3a5eb371ea1027b7.